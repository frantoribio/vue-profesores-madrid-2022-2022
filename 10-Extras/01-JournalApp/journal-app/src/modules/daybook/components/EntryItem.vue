<template>
  <div
    class="m-2 mb-3 cursor-pointer rounded-lg border border-primary bg-base-100 p-2 shadow-md hover:bg-base-200"
  >
    <!-- Titulo -->
    <div class="flex">
      <span
        data-testid="entry-day"
        class="text-lg font-bold text-primary"
        >{{ entryDate.day }}</span
      >
      <span
        data-testid="entry-month"
        class="mx-2 text-lg font-semibold text-base-content"
        >{{ entryDate.month }}</span
      >
      <span
        data-testid="entry-year"
        class="mx-1 text-lg font-light text-base-content"
        >{{ entryDate.year }},
      </span>
      <span
        data-testid="entry-dayofweek"
        class="mx-1 text-lg font-light text-base-content"
        >{{ entryDate.dayOfWeek }}.</span
      >
    </div>
    <!-- Descripcion -->
    <div
      data-testid="entry-text"
      class="text-sm text-base-content"
    >
      {{ shortText }}
    </div>
  </div>
</template>

<script setup>
  import getDateParsered from '@/helpers/dateparser.js'
  import { computed } from 'vue'

  const props = defineProps({
    entry: {
      type: Object,
      required: true,
    },
  })

  const shortText = computed(() => {
    return props.entry.text.length > 130
      ? `${props.entry.text.substring(0, 130)}...`
      : props.entry.text
  })

  const entryDate = computed(() => {
    return getDateParsered(props.entry.date)
  })
</script>

<style lang="scss" scoped></style>
