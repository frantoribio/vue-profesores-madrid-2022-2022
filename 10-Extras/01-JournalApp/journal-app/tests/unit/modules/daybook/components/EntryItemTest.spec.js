import { beforeEach, describe, expect, test } from 'vitest'

import EntryItem from '@/modules/daybook/components/EntryItem.vue'

import { shallowMount } from '@vue/test-utils'

// Suite de Test
describe('Daybook Componente -> EntryItem', () => {
  let wrapper

  const entry = {
    id: '1', // 123456789
    date: new Date().toDateString(), // Sun Dec 16 2019
    text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
    picture: 'https://picsum.photos/600/600', // 'https://picsum.photos/200/300'
  }

  beforeEach(() => {
    wrapper = shallowMount(EntryItem, {
      props: {
        entry,
      },
    })
  })

  test('Debe hacer match con el snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  // Propiedades computadas
  test('Debe tener una propiedad "entry"', () => {
    expect(wrapper.props().entry).toStrictEqual(entry)
  })

  test('Debe tener un texto de como mÃ¡ximo 130 renderizado"', () => {
    const text = wrapper.find('[data-testid="entry-text"]')
    expect(text.text()).toBe(`${entry.text.substring(0, 130)}...`)
  })

  test('Debe renderizar bien la fecha"', () => {
    const entryDate = wrapper.vm.entryDate

    const entryDay = wrapper.find('[data-testid="entry-day"]')
    expect(entryDay.text()).toBe(String(entryDate.day))

    const entryMonth = wrapper.find('[data-testid="entry-month"]')
    expect(entryMonth.text()).toBe(entryDate.month)

    const entryYear = wrapper.find('[data-testid="entry-year"]')
    expect(entryYear.text()).toContain(entryDate.year)
  })
})
