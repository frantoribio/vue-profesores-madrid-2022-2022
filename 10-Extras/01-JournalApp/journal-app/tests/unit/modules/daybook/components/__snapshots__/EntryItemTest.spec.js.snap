// Vitest Snapshot v1

exports[`Daybook Componente -> EntryItem > Debe hacer match con el snapshot 1`] = `
"<div class=\\"m-2 mb-3 cursor-pointer rounded-lg border border-primary bg-base-100 p-2 shadow-md hover:bg-base-200\\">
  <!-- Titulo -->
  <div class=\\"flex\\"><span data-testid=\\"entry-day\\" class=\\"text-lg font-bold text-primary\\">19</span><span data-testid=\\"entry-month\\" class=\\"mx-2 text-lg font-semibold text-base-content\\">Octubre</span><span data-testid=\\"entry-year\\" class=\\"mx-1 text-lg font-light text-base-content\\">2022, </span><span data-testid=\\"entry-dayofweek\\" class=\\"mx-1 text-lg font-light text-base-content\\">Mi√©rcoles.</span></div><!-- Descripcion -->
  <div data-testid=\\"entry-text\\" class=\\"text-sm text-base-content\\">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut eni...</div>
</div>"
`;
